ext/datadog_profiling_native_extension/setup_signal_handler.h

```diff
@@ -2,9 +2,12 @@
 
 #include <signal.h>
 #include "datadog_ruby_common.h"
+#include "ruby_helpers.h"
 
 void empty_signal_handler(DDTRACE_UNUSED int _signal, DDTRACE_UNUSED siginfo_t *_info, DDTRACE_UNUSED void *_ucontext);
-void install_sigprof_signal_handler(void (*signal_handler_function)(int, siginfo_t *, void *), const char *handler_pretty_name);
+#define install_sigprof_signal_handler(signal_handler_function) \
+  private_install_sigprof_signal_handler(signal_handler_function, SIGNAL_HANDLER_NAME(signal_handler_function))
+void private_install_sigprof_signal_handler(void (*signal_handler_function)(int, siginfo_t *, void *), signal_handler_name_t handler_pretty_name);
```

Comment:
The magic is cool but... To be honest in this case I'd go for explicitly passing in the `signal_handler_name_t` entry (even if it's a bit verbose) over having the fancier version (C being already confusing as it is...)